P0009/P0009r12.md

```diff
@@ -708,89 +709,180 @@ properties will work.
+Giving a `gslice` to `valarray::operator[]` returns something that references
+a subset of elements of the original `valarray`.
+However, the result (a `gslice_array` in the nonconst case,
+some type that might be an expression template in the const case)
+is not guaranteed to have an `operator[]`.
+Thus, it's not a view, whereas our proposed `submdspan` function
+always takes and returns a view.
+In the const case, the result might even be a (deep) copy of the input.
+Finally, `gslice` offers no efficient way to address a single element.
+The `gslice` constructor takes strides and lengths as `valarray`s
+and is meant for array-based computation.
+
+## Mixing compile-time and run-time dimensions
+
+The fundamental reason to allow for compile-time dimensions is performance.
 Knowing an extent at compile time enables a number of compiler optimizations,
-such as unrolling and pre-computing of offsets, which can significantly 
-improve the generated code.
-Furthermore in a number of areas some dimensions are more naturally expressed via 
-compile time extents. 
-For example in many physics and engineering algorithms some dimensions are dictated 
-by fundamental properties of the physical world or the discretization scheme.
-The position of a particle in space requries a vector of extents three, since 
-physical space is three-dimensional.
-At the same time other extents are only known at runtime, such as the number of 
-particles in a simulation. 
-Similarly in Graphics, some of the most fundamental objects are 2x2 or 3x3 matrices. 
-Not carrying around the extents as members is important to conserve register and stack space.
-
-<b>Customizable Layout Mappings</b>
-
-One important aspect of allowing customizable layouts is language interoperability.
-C++ arrays have a prescribed layout in memory, with the rightmost index providing stride-1 access
-to the underlying memory. Fortran arrays also have a prescribed layout, however its leftmost
-dimension is providing stride-1 access in memory, the opposite of the C layout.
-Python's NumPy arrays have a configurable layout, to provide compatibility with both C and Fortran.
-
-Beyond language interoperability controling the layout can also be used to enable performant memory
-access patterns which may depend on the hardware. Consider the following small code which performs a
-matrix-vector product using parallel algorithms:
+such as unrolling and pre-computing of offsets.
+These can significantly improve the generated code.
+Not storing dimensions at run time
+may help conserve registers and stack space.
+
+In many fields, some dimensions are naturally known at compile time.
+For many physics and engineering algorithms,
+some dimensions are dictated by fundamental properties of the physical world
+or the discretization scheme.
+For example, the position of a particle in space
+requires a three-dimensional array, since physical space has three dimensions.
```

Comment:
don't say three-dimensional array here. I very explicit used "array of extent three" so it doesn't get confused with the other use of dimension. In particular her you say we represent a "three-dimensional array" with "other dimensions" as a "two-dimensional array"

Reply:
Fixed; thanks!